### PR TITLE
Make sure that test queries process less input bytes with work processor pipelines

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchConnectorFactory.java [hunk 1]
```diff
@@ -20,19 +20,22 @@
...
 
 import static com.google.common.base.MoreObjects.firstNonNull;
+import static java.lang.String.format;
 
 public class TpchConnectorFactory
         implements ConnectorFactory
 {
     public static final String TPCH_COLUMN_NAMING_PROPERTY = "tpch.column-naming";
+    public static final String TPCH_MAX_ROWS_PER_PAGE_PROPERTY = "tpch.max-rows-per-page";
+    private static final int DEFAULT_MAX_ROWS_PER_PAGE = 1000000;
```
Comment (dain) [inline presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchConnectorFactory.java]: Add underscores to the number

File: presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchConnectorFactory.java [hunk 2]
```diff
@@ -95,9 +98,9 @@ public ConnectorSplitManager getSplitManager()
             }
 
             @Override
-            public ConnectorRecordSetProvider getRecordSetProvider()
+            public ConnectorPageSourceProvider getPageSourceProvider()
             {
-                return new TpchRecordSetProvider();
+                return new TpchPageSourceProvider(getMaxRowsPerPage(properties));
```
Comment (dain) [inline presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchConnectorFactory.java]: I'm thinking we might want to make this a configuration option.  In the old code it tested the connector record set code that has a special test for `RecordPageSource` in Presto main that trigger different behavior.

File: presto-tests/src/test/java/io/prestosql/tests/TestWorkProcessorPipelineQueries.java [hunk 3]
```diff
@@ -46,6 +46,7 @@ protected TestWorkProcessorPipelineQueries()
     public void testTopN()
     {
         assertLazyQuery("SELECT * FROM orders ORDER BY totalprice LIMIT 10");
+        assertLazyQuery("SELECT * FROM orders WHERE orderkey >= 0 ORDER BY totalprice LIMIT 10");
```
Comment (dain) [inline presto-tests/src/test/java/io/prestosql/tests/TestWorkProcessorPipelineQueries.java]: IIRC all order keys are positive, so one day this filter could be eliminated by table constraints.  Maybe pick a value like `10`?

File: presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchConnectorFactory.java (hunk 2 above)
Comment (sopel39) [reply presto-tpch/src/main/java/io/prestosql/plugin/tpch/TpchConnectorFactory.java]: I will make record set the default behaviour

**Pull request merged**